### PR TITLE
Subscription management: queue resubscribe to fire after unsubscribe succeeded

### LINKED ISSUE #78626: Subscription Management: Make unsubscribed notice show up faster.
When a user unsubscribes from a site, it takes a while before the notice shows up. This might be technically challenging.
Whereas in Reader `/manage/following`, the toggle is fast, and we need to make sure we can provide the same level of experience.

image

### PR DESCRIPTION
**Pull request by TimBroddin**
Closes #78626

## Proposed Changes

This PR changes the behavior of the unsubscribe toast. Clicking Resubscribe will now only fire after the Unsubscribe mutation has succeeded.

`useRef` is used instead of `useState` because the callback function would otherwise capture the state at the moment of invocation.

## Testing Instructions

- Apply this PR
- Go to `http://calypso.localhost:3000/subscriptions/sites`
- Unsubscribe from a site & resubscribe quickly
- Refresh, the site should still be there

To more easily test this, add this before the API call in `packages/data-stores/src/reader/mutations/use-site-unsubscribe-mutation.ts` (line 42):

`await new Promise(r => setTimeout(r, 10000));`

This will introduce a 10s timeout.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/subscriptions/components/site-subscriptions-manager/site-row.tsx [hunk 1]
```diff
@@ -137,7 +140,11 @@ const SiteRow = ( {
 					duration: 5000,
 					button: translate( 'Resubscribe' ),
 					onClick: () => {
-						resubscribe( { blog_id, url, doNotInvalidateSiteSubscriptions: true } );
+						if ( unsubscribeInProgress.current ) {
+							resubscribePending.current = true;
+						} else {
+							resubscribe( { blog_id, url, doNotInvalidateSiteSubscriptions: true } );
+						}
 						recordSiteResubscribed( {
```
Comment (phcp) [inline client/landing/subscriptions/components/site-subscriptions-manager/site-row.tsx:145]: I noticed the resubscribe event is being tracked for every click, is it expected?
image

**Review by phcp — APPROVED** [review]
Good job @TimBroddin! The PR works as expected. Just left one comment, but if it is an expected behavior feel free to move forward and merge the PR.

**Comment by TimBroddin** [comment]
Closing this PR. Something went wrong during rebase. Will make a new one.

**Comment by jsnajdr** [comment]
> Something went wrong during rebase. Will make a new one.

You could just create a new local `add/subscribers-page-resubscribe-queue` branch, from scratch, totally unrelated to the old one, and force-push it to the GitHub repo. There shouldn't ever be a reason to close the PR. Only the Git branch is in non-recoverable state, the PR is OK.

**Comment by TimBroddin** [comment]
> > Something went wrong during rebase. Will make a new one.
> 
> You could just create a new local `add/subscribers-page-resubscribe-queue` branch, from scratch, totally unrelated to the old one, and force-push it to the GitHub repo. There shouldn't ever be a reason to close the PR. Only the Git branch is in non-recoverable state, the PR is OK.

Thanks for the pointer! I'll do this next time!

**Pull request closed without merging**